### PR TITLE
feat(files): [#FOR-592] send notification when infected file is upload

### PR DESCRIPTION
**Pull request by qperie**
## Describe your changes
Catch infected file alert and send a notification to all the contibutors of the form when it happens.

## Issue ticket number and link
FOR-592 : https://jira.support-ent.fr/browse/FOR-592

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by BastienBenard — COMMENTED** [review]
Il manque juste des tests sur les services

File: formulaire/src/main/java/fr/openent/formulaire/service/impl/DefaultFormService.java [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+import fr.openent.form.core.models.ShareMember;
 import fr.openent.form.helpers.FutureHelper;
 import fr.openent.form.helpers.I18nHelper;
+import fr.openent.form.helpers.IModelHelper;
 import fr.openent.formulaire.service.FormService;
 import fr.wseduc.webutils.Either;
 import io.vertx.core.Future;
 import io.vertx.core.Handler;
 import io.vertx.core.Promise;
+import io.vertx.core.json.Json;
```
Comment (JinDouang) [inline formulaire/src/main/java/fr/openent/formulaire/service/impl/DefaultFormService.java:14]: Import non utilisé

File: formulaire/src/main/java/fr/openent/formulaire/service/impl/FormulaireApplicationStorage.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static fr.openent.form.core.constants.Constants.FORMULAIRE;
+import static fr.openent.form.core.constants.Fields.*;
+import static fr.openent.form.core.constants.Fields.BODY;
+import static fr.openent.form.core.constants.Tables.*;
+
+public class FormulaireApplicationStorage extends PostgresqlApplicationStorage {
+    private final static Logger log = LoggerFactory.getLogger(FormulaireApplicationStorage.class);
```
Comment (JinDouang) [inline formulaire/src/main/java/fr/openent/formulaire/service/impl/FormulaireApplicationStorage.java:33]:
```suggestion
    private static final Logger log = LoggerFactory.getLogger(FormulaireApplicationStorage.class);
```
spec' java sur les Modifiers (les classes, attributs etc..)

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by JinDouang — APPROVED** [review]

File: formulaire/src/main/java/fr/openent/formulaire/service/impl/FormulaireApplicationStorage.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    private final FileInfos fileInfos;
+
+    public FormulaireApplicationStorage(TimelineHelper timelineHelper, EventBus eb) {
+        super(DB_SCHEMA, RESPONSE_FILE_TABLE);
+        this.timelineHelper = timelineHelper;
+        this.formService = new DefaultFormService();
+        this.fileInfos = new FileInfos();
+    }
+
+    protected void getInfoProcess(final String fileId, final Handler<AsyncResult<FileInfos>> handler) {
```
Comment (BastienBenard) [inline formulaire/src/main/java/fr/openent/formulaire/service/impl/FormulaireApplicationStorage.java:45]: @Override

File: formulaire/src/test/java/fr/openent/formulaire/service/test/impl/DefaultFolderServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+        String expectedQuery = "SELECT * FROM " + FOLDER_TABLE + " WHERE id = ?;";
+        JsonArray expectedParams = new JsonArray("[\"1\"]");
+
+        vertx.eventBus().consumer(FORMULAIRE_ADDRESS, message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals(PREPARED, body.getString(ACTION));
+            ctx.assertEquals(expectedQuery, body.getString(STATEMENT));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray(VALUES).toString());
+            async.complete();
```
Comment (BastienBenard) [inline formulaire/src/test/java/fr/openent/formulaire/service/test/impl/DefaultFolderServiceTest.java:55]: Pas nécessaire, car tu l'a dans ton onSuccess

File: formulaire/src/test/java/fr/openent/formulaire/service/test/impl/DefaultFormServiceTest.java [hunk 5]
```diff
@@ -117,6 +117,53 @@ public void testListForLinkerWithGroupsAndUser(TestContext ctx) {
...
+                "UNION " +
+                "SELECT owner_id FROM "  + FORM_TABLE + " WHERE id = ?;";
+        JsonArray expectedParams = new JsonArray("[\"1\", \"fr-openent-formulaire-controllers-FormController|initContribResourceRight\", \"1\"]");
+
+        vertx.eventBus().consumer(FORMULAIRE_ADDRESS, message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals(PREPARED, body.getString(ACTION));
+            ctx.assertEquals(expectedQuery, body.getString(STATEMENT));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray(VALUES).toString());
+            async.complete();
```
Comment (BastienBenard) [inline formulaire/src/test/java/fr/openent/formulaire/service/test/impl/DefaultFormServiceTest.java:136]: Pas nécessaire, car tu l'a dans ton onSuccess

File: formulaire/src/test/java/fr/openent/formulaire/service/test/impl/DefaultFormServiceTest.java [hunk 6]
```diff
@@ -140,6 +187,27 @@ public void testListSentFormsOpeningToday(TestContext ctx) {
...
+
+        String expectedQuery = "SELECT * FROM " + FORM_TABLE + " WHERE id = ?;";
+        JsonArray expectedParams = new JsonArray("[\"1\"]");
+
+        vertx.eventBus().consumer(FORMULAIRE_ADDRESS, message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals(PREPARED, body.getString(ACTION));
+            ctx.assertEquals(expectedQuery, body.getString(STATEMENT));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray(VALUES).toString());
+            async.complete();
```
Comment (BastienBenard) [inline formulaire/src/test/java/fr/openent/formulaire/service/test/impl/DefaultFormServiceTest.java:202]: Pas nécessaire, car tu l'a dans ton onSuccess

**Review by BastienBenard — APPROVED** [review]

**Pull request merged**